src/CodeGen_Hexagon.cpp

```diff
@@ -1297,8 +1312,17 @@ void CodeGen_Hexagon::visit(const Div *op) {
 }
 
 void CodeGen_Hexagon::visit(const Cast *op) {
-    // TODO: Do we need to handle same-sized vector casts before LLVM sees them?
-    CodeGen_Posix::visit(op);
+    if (op->value.type().bits() == 1 && op->type.bits() > 1) {
+        // If left to LLVM, this maps true to one. However,
+        // eliminate_bool_vectors assumes that it maps to negative
+        // one. This will map true to negative one instead.
+        value = call_intrin(op->type,
```

Comment:
... and finding a new way to implement boolean vectors on OpenCL and Hexagon :)

This was an open bug for a very long time before I implemented the eliminate bool vectors stuff in OpenCL, and I remember struggling with finding a better way to do it.

The only way I can see it do it is to add a new type code Bool, so you can have a vector of Bool(32). That seems like a heavy price to pay for this...